cmd/copy.go

```diff
@@ -509,6 +511,19 @@ func (raw rawCopyCmdArgs) cookWithId(jobId common.JobID) (cookedCopyCmdArgs, err
 	}
 	cooked.blobTags = blobTags
 
+	// Check if user has provided `s2s-preserve-blob-tags` flag. If yes, we have to ensure that
+	// 1. Both source and destination must be blob storages.
+	// 2. `blob-tags` is not present as they create conflicting scenario of whether to preserve blob tags from the source or set user defined tags on the destination
+	if raw.s2sPreserveBlobTags {
+		if cooked.fromTo.From() != common.ELocation.Blob() || cooked.fromTo.To() != common.ELocation.Blob() {
+			return cooked, errors.New("either source or destination is not a blob storage. blob index tags is a property of blobs only therefore both source and destination must be blob storage")
```

Comment:
I don't feel like that's necessary, however this can be rephrased though-- "s2s-preserve-blob-tags is only compatible with a blob to blob S2S transfer."

---
In reply to: [568035725](https://github.com/Azure/azure-storage-azcopy/pull/1280#discussion_r568035725) [](ancestors = 568035725)